### ISSUE TITLE
Antialias / Hinting issue

### ISSUE THREAD

**Issue by wez** [linebender/tiny-skia#35]
I'm playing with this crate.  In this particular scenario I'm rendering a horizontal line into a pixmap that is 8 pixels wide and 16 pixels high.  I set the stroke width to 1 pixel, and the line y coordinate is 8.

The resultant bitmap data is:

```
> 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
> 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
> 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
> 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
> 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
> 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
> 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
> 7f7f7f7f 7f7f7f7f 7f7f7f7f 7f7f7f7f 7f7f7f7f 7f7f7f7f 7f7f7f7f 7f7f7f7f
> 80808080 80808080 80808080 80808080 80808080 80808080 80808080 80808080
> 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
> 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
> 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
> 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
> 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
> 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
> 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
```

which looks like the line might be rendering at y = 7.something.  This renders as a fainter grey
color on screen, which is undesirable.

If I turn off anti-aliasing then the line renders at y = 8 at full intensity.

Is the anti-aliasing behavior a bug, or do I need to do something to manually hint the y position (perhaps offsetting by half the stroke width)?

**Comment by RazrFalcon** [comment]
This is the expected behaviour. Skia produces exactly the same output.

I don't think it can be "fixed".

**Comment by ANtlord** [comment]
Hello,

I've been wondering how I can deal with it.  My use case is simple.  I just want to to draw a button with rounded corners.  I did two things:
1. I did 2 instances of `Paint`.  One for the lines, and one for the corners.  The paint for the lines has anti-aliasing disabled, the paint has it enabled.
2. I manually adjusted every corner.  Every corner is a 1/4 of a circle.  Given that, I adjusted the center circle especially for every corner.
The picture looks nice.   However, it falls apart if I make the stroke 2 pixels wide.

I found on the internet, that people draw at 4X scale and then scale down.  Here is [one](https://stackoverflow.com/questions/42465758/about-skia-antialias) of the links. However, this approach doesn't help dealing with fainter grey lines.  Therefore, is there approach which I can use to achieve the goal with this library?
